windows build environment document updates (#692)
ROCm/TheRock: docs/development/windows_support.md

  [Windows Terminal](https://learn.microsoft.com/en-us/windows/terminal/)
  and [Cmder](https://cmder.app/).

> #### Install tools with package manager

Reviewer: nit: adjust grammar in section header and open us up to using other package managers (e.g. WinGet, scoop)
```suggestion
#### Installing tools with the choco package manager
```

> The [choco package manager](https://chocolatey.org/install) from Chocolatey provides an easy method to install TheRock build requirements.
> Below is the commands that will install the tools required by TheRock build.

Reviewer: At a minimum fix the "Below is the commands" grammar (mismatched singular/plural).

Suggested rewording:

```suggestion
The [choco package manager](https://chocolatey.org/install) from Chocolatey provides an easy method to install TheRock build requirements.
These commands can be used to install the tools required for building:
```

#### Install tools

> Chocolatey provides provides [/choco](https://chocolatey.org/install) package manager for Windows which is an easy tool to install the build environment.
> Below is the commands that will install the tools required by TheRock build. Python runtime version it install is 3.13.

Reviewer: Suggested rewording:
```suggestion
The [choco package manager](https://chocolatey.org/install) from Chocolatey can be used to install most tool requirements:
```

If we want a specific version of Python, we can set `choco install python --version=(VERSION) -y` (maybe "3.12"?). Leaving the version unspecified put saying what it _currently_ installs will lead to the docs going stale eventually.

Author: I first tried to use winget instead of installing everything as I though choco was non-free but end having a broken build environment. I think everything else installed fine except the visual studio build tools which end up missing for example the vcvars64.bat. I did not got that to work even afterward by running the vc_Buildtools.exe and selecting the install repair. That caused a broken build as gcc was found on my env instead of microsoft compilers and instead of building  a zlib.dll expected by therock build, I had libzlib.dll, etc...

After I uninstalled everything I had installed with winreg and just used the choco build command everything started working smoothly. Choco is installing python 3.13 and does not support the install of older versions like 3.11. (Found a bug report which says that they do not want to maintain older versions like 3.11)

It seems that 3.12 version could be installed with command with some kind of work-around command: "choco install python312" as standard "choco install python --version 3.12" does not work. So therefore I though it's good to mention that it will now install 3.13 in our build environment by default. (And maybe after some months later some newer version that can then cause problems for doing a build...)

> choco install strawberryperl -y
> ```

Choco will install for now by default the Python version 3.13. If you want to use older python version
like 3.11 that is not maintained by the choco, you will need to use choco to uninstall the python version provided by the choco and then
use some other method to install older python version. For example:

> ```bash
>choco uninstall python
>winget install --id Python.Python.3.11 -e

> > ```

Reviewer: As stated before - avoid writing documentation that we know will get outdated. We aren't that sensitive to the Python version, so I would leave mentions of it out entirely, or specify a minimum like 3.11+.

Keep docs focused on what is essential. We don't want to overload users/developers with information that they don't need.

#  -DCMAKE_C_COMPILER_LAUNCHER=ccache \
#  -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
#  -DCMAKE_MSVC_DEBUG_INFORMATION_FORMAT=Embedded \

> # If wanting to use the amdgpu-windows-interop library from non-default location, you can set:
> -DTHEROCK_AMDGPU_WINDOWS_INTEROP_DIR=<windows-path-to>/amdgpu-windows-interop

Reviewer: nit: match style from the previous block

```suggestion
# To use the amdgpu-windows-interop library from a non-default location, you can set:
#   -DTHEROCK_AMDGPU_WINDOWS_INTEROP_DIR=<windows-path-to>/amdgpu-windows-interop \
```

Author: Yes, this is better wording. Hopefully we can get rid of this eventually by implementing the #693.

#  -DCMAKE_C_COMPILER_LAUNCHER=ccache \
#  -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
#  -DCMAKE_MSVC_DEBUG_INFORMATION_FORMAT=Embedded \

> To use the amdgpu-windows-interop library from a non-default location, you can set:
> #   -DTHEROCK_AMDGPU_WINDOWS_INTEROP_DIR=<windows-path-to>/amdgpu-windows-interop \

Reviewer: My formatting suggestion here was deliberate. Keep the `#` comment character on the `To use...` line since this code block is using bash syntax highlighting and github adds "copy text" button to code blocks, so it should be valid bash still: 
```suggestion
# To use the amdgpu-windows-interop library from a non-default location, you can set:
#   -DTHEROCK_AMDGPU_WINDOWS_INTEROP_DIR=<windows-path-to>/amdgpu-windows-interop \
```
![image](https://github.com/user-attachments/assets/ca7021d3-a381-46c8-9811-569237295455)
